unblob/handlers/archive/instar/hd.py

```diff
@@ -0,0 +1,105 @@
+import io
+import struct
+from pathlib import Path
+from typing import Optional
+
+from structlog import get_logger
+
+from unblob.file_utils import InvalidInputFormat, iterate_patterns
+from unblob.models import Extractor, File, HexString, StructHandler, ValidChunk
+
+logger = get_logger()
+
+EOCD_RECORD_HEADER = 0x9054B50
+
+
+class InstarHDExtractor(Extractor):
+    def extract(self, inpath: Path, outpath: Path):
+        with open(inpath, "rb") as file:
+            outfile = outpath.joinpath(f"{inpath.name}")
+            with open(outfile, "wb") as out_file:
+                headers = file.read()
+
+                if b"\x50\x4b\x03\x07" in headers:
+                    headers = headers.replace(b"\x50\x4b\x03\x07", b"\x50\x4b\x03\x04")
+
+                if b"\x50\x4b\x05\x09" in headers:
+                    headers = headers.replace(b"\x50\x4b\x05\x09", b"\x50\x4b\x05\x06")
+
+                if b"\x50\x4b\x01\x08" in headers:
+                    headers = headers.replace(b"\x50\x4b\x01\x08", b"\x50\x4b\x01\x02")
+                out_file.write(headers)
+
+
+class ZIPHandler(StructHandler):
+    NAME = "hd"
+
+    PATTERNS = [HexString("50 4b 03 07")]  # match on the modified zip header
+
+    C_DEFINITIONS = r"""
+        typedef struct cd_file_header {
+            uint32 magic;
+            uint16 version_made_by;
+            uint16 version_needed;
+            uint16 flags;
+            uint16 compression_method;
+            uint16 dostime;
+            uint16 dosdate;
+            uint32 crc32_cs;
+            uint32 compress_size;
+            uint32 file_size;
+            uint16 file_name_length;
+            uint16 extra_field_length;
+            uint16 file_comment_length;
+            uint16 disk_number_start;
+            uint16 internal_file_attr;
+            uint32 external_file_attr;
+            uint32 relative_offset_local_header;
+            char file_name[file_name_length];
+            char extra_field[extra_field_length];
+        } cd_file_header_t;
+
+        typedef struct end_of_central_directory
+        {
+            uint32 end_of_central_signature;
+            uint16 disk_number;
+            uint16 disk_number_with_cd;
+            uint16 disk_entries;
+            uint16 total_entries;
+            uint32 central_directory_size;
+            uint32 offset_of_cd;
+            uint16 comment_len;
+            char zip_file_comment[comment_len];
+        } end_of_central_directory_t;
+    """
+
+    HEADER_STRUCT = "end_of_central_directory_t"
+
+    EXTRACTOR = InstarHDExtractor()
+
+    def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]:
```

Comment:
this is almost the same as the ZIP handler, can not we inherit from that handler in order to avoid code duplication?